daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/unparsers/UState.scala

```diff
@@ -604,6 +621,23 @@ final class UStateMain private (
     }
   }
 
+  final override def pushTRD(rd: RuntimeData) = {
+    rd match {
+      case trd: TermRuntimeData => inputter.pushTRD(trd)
+      case _ => // ok
+    }
+  }
+  final override def popTRD(rd: RuntimeData) = {
+    rd match {
+      case trd: TermRuntimeData => {
+        val poppedTRD = inputter.popTRD()
+        Assert.invariant(poppedTRD eq rd)
```

Comment:
PUt in better assertion that prints out the two trds so that we have a clue if this assertion in fact fails.